File: backend/RestaurantAPI/Models/Order.cs
namespace RestaurantAPI.Models
{
    public class Order
    {
        public int OrderId { get; set; }
        public int CustomerId { get; set; }
        public DateTime OrderDate { get; set; }
        public string Status { get; set; } = "Pending";
        public decimal TotalAmount { get; set; }

        public Customer Customer { get; set; } = null!;
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}

File: frontend/restaurant-app/src/app/models/order.ts
export interface OrderItem {
  orderItemId?: number;
  menuItemId: number;
  menuItemName?: string;
  quantity: number;
  price?: number;
}

export interface Order {
  orderId: number;
  customerId: number;
  orderDate: string;
  status: string;
  totalAmount: number;
  items: OrderItem[];
}

export interface CreateOrderItem {
  menuItemId: number;
  quantity: number;
}

export interface CreateOrder {
  customerId: number;
  items: CreateOrderItem[];
}

